### ISSUE TITLE
Infinite loop caused by Layout/EndAlignment when EnforcedStyleAlignWith set to variable

### ISSUE THREAD

**Issue by dtcoutu** [rubocop/rubocop#14281]
Received an infinite loop detected error being caused by Layout/EndAlignment.

This came up from an issue reported in [standardrb](https://github.com/standardrb/standard/issues/697).

--------

## Expected behavior

Expected no error and the `end` to be lined up with the beginning of the line. Specifically:

```
    42 + (if @x == 0
            1
          else
            2
    end)
```

## Actual behavior

While the `end` was lined up correctly and infinite loop error was raised.

```
davidcoutu:standardrb-error/ $ be rubocop --debug --config .rubocop.yml --only Layout/EndAlignment -A test.rb                                                                                                          [11:03:21]
Executing bundle exec
configuration from /Users/davidcoutu/work/exploration/standardrb-error/.rubocop.yml
Default configuration from /Users/davidcoutu/.local/share/mise/installs/ruby/3.4.4/lib/ruby/gems/3.4.0/gems/rubocop-1.75.8/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/davidcoutu/work/exploration/standardrb-error/test.rb
Loading cache from /Users/davidcoutu/.cache/rubocop_cache/76070ed566dba0405e079c073161a70773347371/e12e745fe4ca1eab200fc68c20bbed79693b9e35/49e7f851ec7b32674c223519a13bf98c5c49a49e
Infinite loop detected in /Users/davidcoutu/work/exploration/standardrb-error/test.rb and caused by Layout/EndAlignment
W

Offenses:

test.rb:10:5: W: [Corrected] Layout/EndAlignment: end at 10, 4 is not aligned with if at 6, 10.
    end)
    ^^^

1 file inspected, 1 offense detected, 1 offense corrected

1 error occurred:
Infinite loop detected in /Users/davidcoutu/work/exploration/standardrb-error/test.rb and caused by Layout/EndAlignment
configuration from /Users/davidcoutu/work/exploration/standardrb-error/.rubocop.yml
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.75.8 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.45.0, analyzing as Ruby 3.3, running on ruby 3.4.4) [arm64-darwin24]
```

## Steps to reproduce the problem

Created a new test.rb file with:

```
class Test
  def test
    42 + (if @x == 0
            1
          else
            2
    end)
  end
end
```

Then configured the `Layout/EndAlignment` with:
```
Layout/EndAlignment:
  Enabled: true
  EnforcedStyleAlignWith: variable
```

### Notes

Small variations let it work fine:
* if the `end` is lined up with the `if`
* if the `if` is on a new line

## RuboCop version

```
$ bundle exec rubocop -V
1.75.8 (using Parser 3.3.8.0, Prism 1.4.0, rubocop-ast 1.45.0, analyzing as Ruby 3.3, running on ruby 3.4.4) [arm64-darwin24]
```
